step-repositories/step-repositories-artifact/src/main/java/step/repositories/artifact/AbstractArtifactRepository.java

```diff
@@ -111,8 +120,8 @@ public ImportResult importArtefact(ExecutionContext context, Map<String, String>
 		ImportResult result = new ImportResult();
 		List<String> errors = new ArrayList<>();
 		try {
-			ParsedArtifact parsedArtifact = getAndParseArtifact(repositoryParameters);
-			Plan plan = buildTestSetPlan(context, repositoryParameters, parsedArtifact);
+			ParsedArtifact parsedArtifact = getAndParseArtifact(repositoryParameters, context.getObjectEnricher());
+			Plan plan = buildAndStoreTestSetPlan(context, repositoryParameters, parsedArtifact);
```

Comment:
Isolated execution should behave the same whether the artefact is pushed directly from maven plugin/CLI or using an aretafact repository. Wrapping in a test set (ticket SED-3277) will be available for both and optional (false by default).